compiler/rustc_data_structures/src/sync.rs

```diff
@@ -360,6 +413,129 @@ cfg_if! {
     }
 }
 
+pub fn for_each_in<T: IntoIterator>(t: T, mut for_each: impl FnMut(T::Item)) {
+    // We catch panics here ensuring that all the loop iterations execute.
+    // This makes behavior consistent with the parallel compiler.
+    let mut panic = None;
+    t.into_iter().for_each(|i| {
+        if let Err(p) = catch_unwind(AssertUnwindSafe(|| for_each(i))) {
+            if panic.is_none() {
+                panic = Some(p);
+            }
+        }
+    });
+    if let Some(panic) = panic {
+        resume_unwind(panic);
+    }
+}
+
+#[macro_export]
+macro_rules! maybe_parallel {
```

Comment:
All these `maybe` operations should be removed. The runtime branching should happen in the `#[cfg(parallel_compiler)]` path with no `maybe` prefix.

Reply:
Done